docs/contributing.md

```diff
@@ -0,0 +1,267 @@
+  - If you had to upgrade a dependency, why did you do so?
+- **Changelog:** This section is targeted toward consumers — internal developers of the extension or mobile app in addition to external dapp developers — and is intended to be a list of your changes from the perspective of each package in the monorepo. Questions you should seek to answer are:
+  - Which packages are being updated?
+  - What are the _exact_ changes to the API (types, interfaces, functions, methods) that are being changed?
+  - What are the anticipated effects to whichever platform might want to make use of these changes?
+  - If there are breaking changes to the API, what do consumers need to do in order to adapt to those changes upon upgrading to them?
+
+## Testing changes to packages in another project
+
+If you have a project that depends on a package in this monorepo, you may want to load those changes into the project without having to create a whole new monorepo release. How you do this depends on your use case.
+
+### Testing changes to packages locally
+
+If you're developing your project locally and want to test changes to a package, you can follow these steps:
+
+1. First, you must build the monorepo, by running `yarn build`.
+2. Next, you need to connect the package to your project by overriding the resolution logic in your package manager to replace the published version of the package with the local version.
+
+   1. Open `package.json` in the project and locate the dependency entry for the package.
+   2. Locate the section responsible for resolution overrides (or create it if it doesn't exist). If you're using Yarn, this is `resolutions`; if you're using NPM or any other package manager, this is `overrides`.
+   3. Add a line to this section that mirrors the dependency entry on the left-hand side and points to the local path on the right-hand side:
+
+      ```
+      "@metamask/<PACKAGE_NAME>@<PUBLISHED_VERSION_RANGE>": "file:<PATH_TO_METAMASK_DESIGN_SYSTEM>/packages/<PACKAGE_NAME>"
+      ```
+
+      > **Example:**
+      >
+      > - If your project uses Yarn, `@metamask/metamask-module-template` is listed in dependencies at `^1.1.4`, and your clone of the `metamask-design-system` repo is at the same level as your project, add the following to `resolutions`:
+      >
+      >   ```
+      >   "@metamask/metamask-module-template@^1.1.4": "file:../metamask-design-system/packages/metamask-module-template"
+      >   ```
+
+   4. Run `yarn install`.
+
+3. Due to the use of Yarn's `file:` protocol, if you update the package in the monorepo, then you'll need to run `yarn install` in the project again.
+
+### Testing changes to packages with preview builds
+
+If you want to test changes to a package where it would be unwieldy or impossible to use a local version, such as on CI, you can publish a preview build and configure your project to use it.
+
+#### Publishing preview builds as a MetaMask contributor
```

Comment:
Add note that this script doesn't work and that we will be working on a fix and link the issue